server/src/main/java/org/schemarepo/server/JsonRenderer.java

```diff
@@ -0,0 +1,70 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package org.schemarepo.server;
+
+import java.util.Properties;
+
+import javax.ws.rs.core.MediaType;
+
+import org.schemarepo.RepositoryUtil;
+import org.schemarepo.SchemaEntry;
+import org.schemarepo.Subject;
+import org.schemarepo.json.JsonUtil;
+
+/**
+ * Renders as JSON. The actual work is delegated to utility class {@link org.schemarepo.RepositoryUtil}
+ */
+public class JsonRenderer implements Renderer {
+
+  private final JsonUtil jsonUtil;
+
+  public JsonRenderer(JsonUtil jsonUtil) {
+    if (jsonUtil == null) {
+      throw new IllegalArgumentException("jsonUtil required");
+    }
+    this.jsonUtil = jsonUtil;
+  }
+
+  @Override
+  public String getMediaType() {
+    return MediaType.APPLICATION_JSON;
+  }
+
+  @Override
+  public String renderSubjects(Iterable<Subject> subjects) {
+    return jsonUtil.subjectsToJson(subjects);
+  }
+
+  @Override
+  public String renderSchemas(Iterable<SchemaEntry> schemaEntries) {
+    return jsonUtil.schemasToJson(schemaEntries);
+  }
+
+  @Override
+  public String renderSchemaEntry(SchemaEntry schemaEntry, boolean requestForLatest) {
+    return requestForLatest ? schemaEntry.toString() : schemaEntry.getSchema();
+  }
+
+  @Override
+  public String renderProperties(Properties props, String comment) {
+    // TODO: when the need arises, this may yield an actual JSON
```

Comment:
I'm not sure I understand what do you propose.

In order for an endpoint to serve different response depending on the `Accept` header, the corresponding functionality must be provided in the `Renderer` interface. The alternative is to bake the logic ("if mediaType is ...") into `RESTRepository` itself, which I find less attractive stylistically and as far as separation of concerns goes.

In this particular case, I'm saying (through code) that we don't elect to render something in a particular way yet. This is consistent / compatible with the original behavior of completely ignoring `Accept` header. Responding with `415` may be more correct formally, but I don't see why would we want to do that from practical point of view.

I see your concern of "converting" an endpoint to support different views based on the media type as being a little heavy (respective implementations in all classes). I contemplated going down the path of not having a method per endpoint but rather having a generic `render` taking a map of params and an enum for an operation/endpoint, but I thought that would make the code unnecessarily more complex. We have a very limited number of endpoints, most of which already support per-content views, and I don't see how that can grow significantly. Regarding having HTML-only endpoints, I don't quite see how does that fit into existing realm. My goal was to provide quick (and perhaps somewhat dirty) way of accessing the repo without access to underlying backend or having to use the likes of `curl`. It has never involved creating a full-blown website on top of the repo.